systemtests/src/test/java/io/enmasse/systemtest/marathon/RestartTest.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright 2018, EnMasse authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+package io.enmasse.systemtest.marathon;
+
+import io.enmasse.systemtest.*;
+import io.fabric8.kubernetes.api.model.Pod;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.Test;
+import org.slf4j.Logger;
+
+import java.util.List;
+import java.util.Random;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+
+class RestartTest extends MarathonTestBase {
+    private static Logger log = CustomLogger.getLogger();
+    private ScheduledExecutorService deleteService;
+
+    @AfterEach
+    void tearDownRestart() {
+        if (deleteService != null) {
+            deleteService.shutdownNow();
+        }
+    }
+
+    @Test
+    void testRandomDeletePods() throws Exception {
+
+        UserCredentials user = new UserCredentials("test-user", "passsswooooord");
+        AddressSpace standard = new AddressSpace("addr-space-restart-standard", AddressSpaceType.STANDARD, AuthService.STANDARD);
+        AddressSpace brokered = new AddressSpace("addr-space-restart-brokered", AddressSpaceType.BROKERED, AuthService.STANDARD);
+        createAddressSpaceList(standard, brokered);
+        createUser(brokered, user);
+        createUser(standard, user);
+
+        List<Destination> brokeredAddresses = getAllBrokeredAddresses();
+        List<Destination> standardAddresses = getAllStandardAddresses();
+
+        setAddresses(brokered, brokeredAddresses.toArray(new Destination[0]));
+        setAddresses(standard, standardAddresses.toArray(new Destination[0]));
+
+        assertCanConnect(brokered, user, brokeredAddresses);
+        assertCanConnect(standard, user, standardAddresses);
+
+        //set up restart scheduler
+        deleteService = Executors.newSingleThreadScheduledExecutor();
```

Comment:
Set this up in a @BeforeEach setup method instead since it is torn down in aftereach anyway

Reply:
good point